src/routes/middlewares/duplicate-user.middleware.js

```diff
@@ -0,0 +1,63 @@
+import {
+  getUserbyUsername,
+  getUserbyId,
+  errorResponse,
+  checkDuplicateUser,
+} from '../../utils/helpers.utils';
+
+/**
+ *
+ *
+ * @export
+ * @param {object} req express request
+ * @param {object} res express response
+ * @param {void} next handler
+ * @returns {void}
+ */
+export async function usernameChecker(req, res, next) {
+  const { id } = req.user;
+  const { username } = req.body;
+  const user = await getUserbyId(id);
+  const usernameExists = await getUserbyUsername(username);
+
+  if (usernameExists && user.id !== usernameExists.id) {
```

Comment:
The naming of this variable 'usernameExists' got me into thinking it's a boolean datatype, but in reading on, I see it is actually an object.

What do you think if we called it userDetails or some other name that easily depicts it as an object?

Reply:
My thinking on the naming convention is the way it fits with the statement it is used in immediately after it's declared. `if usernameExists` sounds better than `if userDetails`...